src/components/NameValueList/NameValueList.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+export const List = ({
+  children,
+  className,
+  mode,
+  ...rest
+}) => (
+  <dl
+    {...rest}
+    className={cx([
+      `slds-list_${mode}`,
+      { 'slds-wrap': mode === 'horizontal' },
+      className,
+    ])}
+  >
+    {children}
+  </dl>
+);
+
+List.defaultProps = {
+  className: null,
+  mode: 'horizontal',
+};
+
+List.propTypes = {
+  /**
+   * one or more sets of `Value` or `Name` components
+   */
+  children: PropTypes.node.isRequired,
+  className: PropTypes.string,
+  mode: PropTypes.oneOf(['horizontal', 'inline', 'stacked']),
+};
+
+const NameOrValue = ({
+  as: El,
+  className,
+  children,
+  title,
+  ...rest
+}) => (
+  <El
+    {...rest}
+    className={cx([className, 'slds-truncate'])}
+    title={title || (typeof children === 'string' ? children : '')}
+  >
+    {children}
+  </El>
+);
+
+NameOrValue.defaultProps = {
+  className: null,
+  title: null,
+};
+
+NameOrValue.propTypes = {
+  as: PropTypes.string.isRequired,
```

Comment:
Doesn't SF have a limitation on which lags can be used?

Reply:
No they dont

Reply:
In this case, it would make sense to limit to `dt,dd` since we are inside a `dl`. Will add that